src/main/scala/com/fulcrumgenomics/umi/ConsensusCaller.scala

```diff
@@ -81,12 +82,15 @@ class ConsensusCaller(errorRatePreLabeling:  PhredScore,
     */
   class ConsensusBaseBuilder {
     private val observations = new Array[Int](DnaBaseCount)
-    private val likelihoods = new Array[LogProbability](DnaBaseCount)
+
+    // Note: to ensure numerical stability, we store the terms we want to eventually sum, rather than storing the sum
+    // itself. We can then sum smallest (in magnitude) to largest (in magnitude) for numerical stability.
+    private val likelihoods = Range.inclusive(1, DnaBaseCount).map { _ => new ArrayBuffer[LogProbability](256) }.toArray
```

Comment:
I _think_ sorting isn't that bad in terms of increased performance (run time), but if others disagree, how would you feel about merging this PR, then following up with another PR with those changes.

For example, instead a array of all values, since we're converting Phred values, we could have a fixed size array of zeros and increment the offsets based on the Phred value.  Then when we sum, we can instead sort the unique values rather than all the values.  We would convert the phred value to log probability right before the sum.